src/main/java/tutorlink/grade/Grade.java

```
@@ -0,0 +1,26 @@
+package tutorlink.grade;
+
+import tutorlink.component.Component;
+import tutorlink.student.Student;
+
+public class Grade {
+    private double score;
+    private Component component;
+    private Student student;
+
+    public Grade(Component component, Student student, double score) {
+        this.component = component;
+        this.student = student;
+        this.score = score;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof Grade grade) {
+            return this.score == grade.score
```

I think we shouldn't be checking for the equivalence of score because even if the scores are different as long as the component and the student is the same, we should throw an exception (duplicate Grade object). 

Unless we want to implement the functionality where it overrides the existing Grade object with one with a new score?

---

equals shouldn't return exception, though we can make a separate function to check if it's same component same student but different score